src/pnpq/devices/odl_ozoptics_650ml.py

```diff
@@ -151,24 +153,23 @@ def serial_send(self, serial_cmd):
         self.conn.flushOutput()  # flush output buffer, aborting current output and discard all that is in buffer
         self.conn.write(serial_cmd.encode())
 
-    def serial_read(self, retries=10):
+    def serial_read(self):
         # The Python serial "in_waiting" property is the count of bytes available
         # for reading at the serial port.  If the value is greater than zero
         # then we know we have content available.
         device_output = ""
-        got_OK = False
-        while self.conn.in_waiting > 0 or (got_OK is False and retries > 0):
-            device_output += (self.conn.read(self.conn.in_waiting)).decode("iso-8859-1")
-            # command output is complete.
-            if device_output.find("Done") >= 0:
-                got_OK = True
-            time.sleep(0.05)
-            retries -= 1
+        device_output += (self.conn.read_until(expected=bytearray(b"Done"))).decode(
+            "iso-8859-1"
+        )
```

Comment:
You can delete line 160 (`device_output = ""`).

```suggestion
        device_output = self.conn.read_until(expected=bytearray(b"Done")).decode("iso-8859-1")
```

Reply:
Done!